genomic_data_service/region_indexer.py

```diff
@@ -68,68 +73,220 @@
+            continue
+
+        if f['file_format'] != 'bed':
+            log(f"Invalid file format: {f['file_format']}")
+            continue
 
-    for dataset_file in dataset_files:
         requirements = None
-        
+
         for collection_type in REGULOME_COLLECTION_TYPES:
-            if collection_type in dataset_file:
-                requirements = REGULOME_REGION_REQUIREMENTS.get(dataset_file[collection_type].lower())
+            if collection_type in f:
+                if isinstance(f[collection_type], list):
+                    for ctype in f[collection_type]:
+                        if ctype.lower() in REGULOME_REGION_REQUIREMENTS:
+                            requirements = REGULOME_REGION_REQUIREMENTS.get(ctype.lower())
+                            break
+                else:
+                    requirements = REGULOME_REGION_REQUIREMENTS.get(f[collection_type].lower())
 
-        if not requirements:
-            print(f'{dataset_file["@id"]}')
+                if requirements is None:
+                    log(f'Unsupported {collection_type}: {f[collection_type]}')
+
+        if requirements is None:
             continue
 
-        if dataset_file.get('files'):
-            bed_file = dataset_file['files'][0] # TODO: change to preferred_default when ready
+        for requirement in ['output_type', 'file_type', 'file_format']:
+            if requirement in requirements and f[requirement].lower() not in requirements[requirement]:
+                log(f'Requirement {requirement} not satisfied: {f[requirement]} - expected: {requirements[requirement]}')
+                continue
+
+        files_to_index.append(f)
+
+    return files_to_index
+
+
+def dataset_accession(f):
+    if 'dataset' not in f:
+        return None
+
+    return f['dataset'].split('/')[2]
+
+
+def fetch_datasets(files, datasets):
+    fetch = []
+
+    for f in files:
+        accession = dataset_accession(f)
+
+        if accession in datasets:
+            continue
+
+        fetch.append(accession)
+
+    datasets_data = encode_graph([f'accession={c}' for c in fetch])
+
+    for dataset in datasets_data:
+        fetch_documents(dataset)
+        datasets[dataset['accession']] = dataset
+
+
+def index_regulome_db(filter_files=False):
+    regulome_encode_accessions = pickle.load(open(REGULOME_ENCODE_ACCESSIONS_MAPPING_PATH, 'rb'))
+    regulome_accessions = list(pickle.load(open(REGULOME_ACCESSIONS_PATH, 'rb')))
+
+    encode_accessions = [regulome_encode_accessions[reg] for reg in regulome_accessions]
+
+    datasets = {}
+
+    per_request = 350
+    chunks = [encode_accessions[i:i + per_request] for i in range(0, len(encode_accessions), per_request)]
+
+    i = 0
+    print_progress_bar(i, len(chunks))
+    for chunk in chunks:
+        i += 1
+        print_progress_bar(i, len(chunks))
+
+        files = encode_graph([f'accession={c}' for c in chunk])
+
+        files_to_index = filter_files(files) if filter_files else files
```

Comment:
the `filter_files` method was extracted and refactored from the original regulome codebase. It reads all existing files and selects only those eligible to be part of the region search. For now it is disabled as we are reading the list of accessions directly, but it's useful to keep it for the future.